### PR TITLE
[monitoring] add an alert about deprecated OS versions

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
Signed-off-by: Denis Romanenko <denis.romanenko@flant.com>

## Description

In deckhouse release 1.43 we plan to remove Ubuntu 16.04 support.
This PR adds alerting about Ubuntu 16.04 deprecation.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: monitoring
type: chore
summary: Add an alert about deprecated OS versions.
```

### CODE REVIEW

File: modules/340-monitoring-deckhouse/monitoring/prometheus-rules/kernel-requirements.yaml [hunk 1]
```diff
@@ -10,11 +10,9 @@
       annotations:
         plk_markup_format: markdown
         plk_protocol_version: "1"
-        plk_create_group_if_not_exists__cluster_has_problems_with_unmet_kernel_requirements_alerts: "D8ClusterHasProblemsWithUnmetKernelRequirements,tier=cluster,prometheus=deckhouse,kubernetes=~kubernetes"
-        plk_grouped_by__cluster_has_problems_with_unmet_kernel_requirements_alerts: "D8ClusterHasProblemsWithUnmetKernelRequirements,tier=cluster,prometheus=deckhouse,kubernetes=~kubernetes"
-        summary: Node {{ $labels.node }} has unmet kernel requirements
+        summary: Nodes has unmet kernel requirements
         description: |-
-          Some nodes have unmet kernel constraints. This means that some modules cannot be run on that nodes.
+          Some nodes has unmet kernel constraints. This means that some modules cannot be run on that nodes.
```
Comment (apolovov) [inline modules/340-monitoring-deckhouse/monitoring/prometheus-rules/kernel-requirements.yaml:15]:
```suggestion
          Some nodes have unmet kernel constraints. This means that some modules cannot be run on that nodes.
```
It's plural

File: modules/340-monitoring-deckhouse/monitoring/prometheus-rules/kernel-requirements.yaml [hunk 2]
```diff
@@ -10,11 +10,9 @@
       annotations:
         plk_markup_format: markdown
         plk_protocol_version: "1"
-        plk_create_group_if_not_exists__cluster_has_problems_with_unmet_kernel_requirements_alerts: "D8ClusterHasProblemsWithUnmetKernelRequirements,tier=cluster,prometheus=deckhouse,kubernetes=~kubernetes"
-        plk_grouped_by__cluster_has_problems_with_unmet_kernel_requirements_alerts: "D8ClusterHasProblemsWithUnmetKernelRequirements,tier=cluster,prometheus=deckhouse,kubernetes=~kubernetes"
-        summary: Node {{ $labels.node }} has unmet kernel requirements
+        summary: Nodes has unmet kernel requirements
```
Comment (apolovov) [inline modules/340-monitoring-deckhouse/monitoring/prometheus-rules/kernel-requirements.yaml:13]:
```suggestion
        summary: Nodes have unmet kernel requirements
```

File: modules/340-monitoring-deckhouse/monitoring/prometheus-rules/node-os-requirements.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      expr: >-
+        count(kube_node_info{os_image=~"Ubuntu 16.04.*"}) > 0
+      for: 5m
+      labels:
+        severity_level: "4"
+        tier: cluster
+      annotations:
+        plk_markup_format: markdown
+        plk_protocol_version: "1"
+        summary: Nodes has deprecated OS versions.
```
Comment (apolovov) [inline modules/340-monitoring-deckhouse/monitoring/prometheus-rules/node-os-requirements.yaml:13]:
```suggestion
        summary: Nodes have deprecated OS versions.
```

File: modules/340-monitoring-deckhouse/monitoring/prometheus-rules/node-os-requirements.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        description: |-
+          Some nodes has deprecated OS versions. Please update nodes to actual OS version.
```
Comment (apolovov) [inline modules/340-monitoring-deckhouse/monitoring/prometheus-rules/node-os-requirements.yaml:15]:
```suggestion
          Some nodes have deprecated OS versions. Please update nodes to actual OS version.
```

**Review by apolovov — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [3408](https://github.com/deckhouse/deckhouse/pull/3408) to the branch [release-1.42](https://github.com/deckhouse/deckhouse/tree/release-1.42) successful!

**Pull request merged**